orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/Message.kt

```diff
@@ -24,12 +24,50 @@ import com.netflix.spinnaker.orca.Task
 import com.netflix.spinnaker.orca.pipeline.model.Execution
 import com.netflix.spinnaker.orca.pipeline.model.Stage
 import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import java.util.*
 
 /**
  * Messages used internally by the queueing system.
  */
+
+@JsonTypeInfo(use = MINIMAL_CLASS, include = PROPERTY, property = "@class")
+interface Attribute {
+}
+
+data class MaxAttemptsAttribute(val maxAttempts: Int = -1) : Attribute {
+}
+
+data class AttemptsAttribute(var attempts: Int = 0) : Attribute {
+  fun increment() {
+    this.attempts = attempts + 1
+  }
+}
```

Comment:
The primary point, for me, is that we decouple the Attributes that can be assigned to a message from things that might use it. I don't want to update an Attributes class when we implement new / custom rate limiters. For example, the `PriorityCapacity` traffic shaper is basically a dumb interface in OSS-land, but will be expanded dramatically internally for our specific use cases. Being able to add our own custom metadata via explicit and separate, concrete Attribute classes makes sense to me for that use case.